pkg/sql/catalog/replication/replication_test: TestReaderCatalogTSAdvance failed

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://mesolite.cluster.engflow.com/invocations/default/64067bc9-b7db-4689-bf62-b2ad2a948dc0?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvcmVwbGljYXRpb246cmVwbGljYXRpb25fdGVzdA==) on master @ [ef26d1534507f6089aa909e0bd43a0efb71dd6c7](https://github.com/cockroachdb/cockroach/commits/ef26d1534507f6089aa909e0bd43a0efb71dd6c7):


```
I241030 21:41:57.154548 12078 server/start_listen.go:111  [T11,Vdest,n1] 2720  server shutting down: instructing cmux to stop accepting
W241030 21:41:57.154695 12081 kv/kvclient/kvtenant/connector.go:424  [T11,Vdest,n1,tenant-connector] 2721  error consuming GossipSubscription RPC: recv msg error: grpc: context canceled [code 1/Canceled]
W241030 21:41:57.155114 45542 jobs/adopt.go:490  [T11,Vdest,n1] 2722  could not clear job claim: clear-job-claim: node unavailable; try another peer
W241030 21:41:57.155554 45519 jobs/adopt.go:490  [T11,Vdest,n1] 2723  could not clear job claim: clear-job-claim: node unavailable; try another peer
I241030 21:41:57.155873 12263 jobs/registry.go:1598  [T11,Vdest,n1] 2724  AUTO SPAN CONFIG RECONCILIATION job 1016640534474129409: stepping through state succeeded
W241030 21:41:57.156166 12263 jobs/adopt.go:490  [T11,Vdest,n1] 2725  could not clear job claim: clear-job-claim: node unavailable; try another peer
W241030 21:41:57.156276 12082 kv/kvclient/kvtenant/setting_overrides.go:65  [T11,Vdest,n1,tenant-connector] 2726  error consuming TenantSettings RPC: recv msg error: grpc: context canceled [code 1/Canceled]
W241030 21:41:57.156439 12131 sql/sqlliveness/slinstance/slinstance.go:336  [T11,Vdest,n1] 2727  exiting heartbeat loop
W241030 21:41:57.156496 12131 sql/sqlliveness/slinstance/slinstance.go:323  [T11,Vdest,n1] 2728  exiting heartbeat loop with error: node unavailable; try another peer
E241030 21:41:57.156538 12131 server/server_sql.go:515  [T11,Vdest,n1] 2729  failed to run update of instance with new session ID: node unavailable; try another peer
I241030 21:41:57.156711 43303 jobs/registry.go:1598  [T10,Vsrc,n1] 2730  AUTO UPDATE SQL ACTIVITY job 103: stepping through state succeeded
W241030 21:41:57.157012 43303 jobs/adopt.go:490  [T10,Vsrc,n1] 2731  could not clear job claim: clear-job-claim: node unavailable; try another peer
I241030 21:41:57.157089 11816 server/server_controller_channel_orchestrator.go:502  [n1,tenant-orchestration,tenant=src] 2732  propagate-graceful-drain task terminating
I241030 21:41:57.157153 11809 sql/stats/automatic_stats.go:712  [T10,Vsrc,n1] 2733  quiescing stats garbage collector
I241030 21:41:57.157201 11410 server/server_controller_channel_orchestrator.go:442  [n1,tenant-orchestration,tenant=src] 2734  tenant "src" finishing their own control loop
I241030 21:41:57.157337 11468 server/start_listen.go:111  [T10,Vsrc,n1] 2735  server shutting down: instructing cmux to stop accepting
W241030 21:41:57.157613 11513 sql/sqlliveness/slinstance/slinstance.go:336  [T10,Vsrc,n1] 2736  exiting heartbeat loop
W241030 21:41:57.157674 11513 sql/sqlliveness/slinstance/slinstance.go:323  [T10,Vsrc,n1] 2737  exiting heartbeat loop with error: node unavailable; try another peer
E241030 21:41:57.157709 11513 server/server_sql.go:515  [T10,Vsrc,n1] 2738  failed to run update of instance with new session ID: node unavailable; try another peer
W241030 21:41:57.157942 11471 kv/kvclient/kvtenant/connector.go:424  [T10,Vsrc,n1,tenant-connector] 2739  error consuming GossipSubscription RPC: recv msg error: grpc: error reading from server: read tcp 127.0.0.1:37534->127.0.0.1:46593: use of closed network connection [code 14/Unavailable]
I241030 21:41:57.158093 11409 server/server_controller_channel_orchestrator.go:261  [T1,Vsystem,n1] 2740  server terminating; telling tenant "src" to terminate
W241030 21:41:57.158247 11472 kv/kvclient/kvtenant/setting_overrides.go:65  [T10,Vsrc,n1,tenant-connector] 2741  error consuming TenantSettings RPC: recv msg error: grpc: error reading from server: read tcp 127.0.0.1:37534->127.0.0.1:46593: use of closed network connection [code 14/Unavailable]
E241030 21:41:57.158382 43300 jobs/registry.go:897  [T10,Vsrc,n1] 2742  error getting live session: node unavailable; try another peer
I241030 21:41:57.158509 11652 jobs/registry.go:1598  [T10,Vsrc,n1] 2743  AUTO SPAN CONFIG RECONCILIATION job 1016640533010153473: stepping through state succeeded
I241030 21:41:57.159144 10991 sql/stats/automatic_stats.go:712  [T1,Vsystem,n1] 2744  quiescing stats garbage collector
I241030 21:41:57.159343 9654 server/start_listen.go:111  [T1,Vsystem,n1] 2745  server shutting down: instructing cmux to stop accepting
W241030 21:41:57.159808 40751 jobs/adopt.go:490  [T1,Vsystem,n1] 2746  could not clear job claim: clear-job-claim: node unavailable; try another peer
W241030 21:41:57.160547 40774 jobs/adopt.go:490  [T1,Vsystem,n1] 2747  could not clear job claim: clear-job-claim: node unavailable; try another peer
I241030 21:41:57.160656 10559 jobs/registry.go:1598  [T1,Vsystem,n1] 2748  AUTO SPAN CONFIG RECONCILIATION job 1016640530984763393: stepping through state succeeded
W241030 21:41:57.160958 10559 jobs/adopt.go:490  [T1,Vsystem,n1] 2749  could not clear job claim: clear-job-claim: node unavailable; try another peer
W241030 21:41:57.161043 10085 sql/sqlliveness/slinstance/slinstance.go:336  [T1,Vsystem,n1] 2750  exiting heartbeat loop
W241030 21:41:57.161095 10085 sql/sqlliveness/slinstance/slinstance.go:323  [T1,Vsystem,n1] 2751  exiting heartbeat loop with error: node unavailable; try another peer
E241030 21:41:57.161148 10085 server/server_sql.go:515  [T1,Vsystem,n1] 2752  failed to run update of instance with new session ID: node unavailable; try another peer
E241030 21:41:57.161362 40788 jobs/registry.go:897  [T1,Vsystem,n1] 2753  error getting live session: node unavailable; try another peer
I241030 21:41:57.161540 12507 4@util/log/event_log.go:39  [T10,Vsrc,n1,client=127.0.0.1:47334,hostssl,user=root] 2754 ={"Timestamp":1730324517161537763,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:47334","SessionID":"180358effcf29c2d0000000000000001","Duration":47556946955}
I241030 21:41:57.161793 15109 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:47346,hostssl,user=root] 2755 ={"Timestamp":1730324517161792204,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:47346","SessionID":"180358f11ee7792c0000000000000001","Duration":42692539242}
W241030 21:41:57.164193 12035 server/server_sql.go:1780  [T11,Vdest,n1] 2756  server shutdown without a prior graceful drain
I241030 21:41:57.164258 12035 1@util/log/event_log.go:39  [n1,tenant-orchestration,tenant=dest] 2757 ={"Timestamp":1730324517164257027,"EventType":"tenant_shared_service_stop","NodeID":1,"TenantID":11,"InstanceID":1,"TenantName":"dest"}
W241030 21:41:57.164312 12035 sql/event_log.go:658  [n1,tenant-orchestration,tenant=dest] 2758  failed to start task to save 1 events in eventlog: node unavailable; try another peer
I241030 21:41:57.164663 12035 server/server_controller_channel_orchestrator.go:486  [n1,tenant-orchestration,tenant=dest] 2759  propagate-ungraceful-stop task terminating
I241030 21:41:57.164728 12033 server/server_controller_channel_orchestrator.go:286  [T1,Vsystem,n1] 2760  propagate-close task terminating
W241030 21:41:57.164830 11411 server/server_sql.go:1780  [T10,Vsrc,n1] 2761  server shutdown without a prior graceful drain
I241030 21:41:57.164875 11411 1@util/log/event_log.go:39  [n1,tenant-orchestration,tenant=src] 2762 ={"Timestamp":1730324517164874367,"EventType":"tenant_shared_service_stop","NodeID":1,"TenantID":10,"InstanceID":1,"TenantName":"src"}
W241030 21:41:57.164925 11411 sql/event_log.go:658  [n1,tenant-orchestration,tenant=src] 2763  failed to start task to save 1 events in eventlog: node unavailable; try another peer
I241030 21:41:57.165224 11411 server/server_controller_channel_orchestrator.go:486  [n1,tenant-orchestration,tenant=src] 2764  propagate-ungraceful-stop task terminating
I241030 21:41:57.169614 11409 server/server_controller_channel_orchestrator.go:286  [T1,Vsystem,n1] 2765  propagate-close task terminating
I241030 21:41:57.174798 9577 server/server_controller.go:409  [T1,Vsystem,n1] 2766  server controller shutting down
I241030 21:41:57.174858 9577 server/server_controller.go:418  [T1,Vsystem,n1] 2767  waiting for tenant servers to report stopped
W241030 21:41:57.174948 9577 server/server_sql.go:1780  [T1,Vsystem,n1] 2768  server shutdown without a prior graceful drain
--- FAIL: TestReaderCatalogTSAdvance (49.10s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-43814

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://tanzanite.cluster.engflow.com/invocations/default/8d98454b-0bc8-48c3-87ab-780c2a17f020?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvcmVwbGljYXRpb246cmVwbGljYXRpb25fdGVzdA==) on master @ [becbd0fcdfa2e37a6ff23b33af70f2f91eca0790](https://github.com/cockroachdb/cockroach/commits/becbd0fcdfa2e37a6ff23b33af70f2f91eca0790):


```
W241031 08:50:16.585119 12058 kv/kvclient/kvtenant/setting_overrides.go:65  [T11,Vdest,n1,tenant-connector] 2708  error consuming TenantSettings RPC: recv msg error: grpc: context canceled [code 1/Canceled]
W241031 08:50:16.585249 12057 kv/kvclient/kvtenant/connector.go:424  [T11,Vdest,n1,tenant-connector] 2709  error consuming GossipSubscription RPC: recv msg error: grpc: context canceled [code 1/Canceled]
I241031 08:50:16.585538 12256 jobs/registry.go:1598  [T11,Vdest,n1] 2710  AUTO SPAN CONFIG RECONCILIATION job 1016771932963700737: stepping through state succeeded
W241031 08:50:16.585898 12256 jobs/adopt.go:490  [T11,Vdest,n1] 2711  could not clear job claim: clear-job-claim: node unavailable; try another peer
W241031 08:50:16.586280 47906 jobs/adopt.go:490  [T11,Vdest,n1] 2712  could not clear job claim: clear-job-claim: node unavailable; try another peer
W241031 08:50:16.586477 12116 sql/sqlliveness/slinstance/slinstance.go:336  [T11,Vdest,n1] 2713  exiting heartbeat loop
W241031 08:50:16.586527 12116 sql/sqlliveness/slinstance/slinstance.go:323  [T11,Vdest,n1] 2714  exiting heartbeat loop with error: node unavailable; try another peer
E241031 08:50:16.586564 12116 server/server_sql.go:515  [T11,Vdest,n1] 2715  failed to run update of instance with new session ID: node unavailable; try another peer
I241031 08:50:16.586712 49456 jobs/registry.go:1598  [T10,Vsrc,n1] 2716  AUTO UPDATE SQL ACTIVITY job 103: stepping through state succeeded
W241031 08:50:16.587183 49456 jobs/adopt.go:490  [T10,Vsrc,n1] 2717  could not clear job claim: clear-job-claim: node unavailable; try another peer
I241031 08:50:16.587289 11821 server/server_controller_channel_orchestrator.go:502  [n1,tenant-orchestration,tenant=src] 2718  propagate-graceful-drain task terminating
I241031 08:50:16.587376 11382 server/server_controller_channel_orchestrator.go:261  [T1,Vsystem,n1] 2719  server terminating; telling tenant "src" to terminate
I241031 08:50:16.587420 11814 sql/stats/automatic_stats.go:712  [T10,Vsrc,n1] 2720  quiescing stats garbage collector
I241031 08:50:16.587486 11383 server/server_controller_channel_orchestrator.go:442  [n1,tenant-orchestration,tenant=src] 2721  tenant "src" finishing their own control loop
I241031 08:50:16.587648 11434 server/start_listen.go:111  [T10,Vsrc,n1] 2722  server shutting down: instructing cmux to stop accepting
W241031 08:50:16.588400 49441 jobs/adopt.go:490  [T10,Vsrc,n1] 2723  could not clear job claim: clear-job-claim: node unavailable; try another peer
W241031 08:50:16.588870 49475 jobs/adopt.go:490  [T10,Vsrc,n1] 2724  could not clear job claim: clear-job-claim: node unavailable; try another peer
W241031 08:50:16.588950 11437 kv/kvclient/kvtenant/connector.go:424  [T10,Vsrc,n1,tenant-connector] 2725  error consuming GossipSubscription RPC: recv msg error: grpc: context canceled [code 1/Canceled]
W241031 08:50:16.589087 11438 kv/kvclient/kvtenant/setting_overrides.go:65  [T10,Vsrc,n1,tenant-connector] 2726  error consuming TenantSettings RPC: recv msg error: grpc: context canceled [code 1/Canceled]
I241031 08:50:16.589312 11626 jobs/registry.go:1598  [T10,Vsrc,n1] 2727  AUTO SPAN CONFIG RECONCILIATION job 1016771931317239809: stepping through state succeeded
W241031 08:50:16.589661 11626 jobs/adopt.go:490  [T10,Vsrc,n1] 2728  could not clear job claim: clear-job-claim: node unavailable; try another peer
W241031 08:50:16.589789 11479 sql/sqlliveness/slinstance/slinstance.go:336  [T10,Vsrc,n1] 2729  exiting heartbeat loop
W241031 08:50:16.589873 11479 sql/sqlliveness/slinstance/slinstance.go:323  [T10,Vsrc,n1] 2730  exiting heartbeat loop with error: node unavailable; try another peer
E241031 08:50:16.589915 11479 server/server_sql.go:515  [T10,Vsrc,n1] 2731  failed to run update of instance with new session ID: node unavailable; try another peer
E241031 08:50:16.590047 49465 jobs/registry.go:897  [T10,Vsrc,n1] 2732  error getting live session: node unavailable; try another peer
I241031 08:50:16.590449 10966 sql/stats/automatic_stats.go:712  [T1,Vsystem,n1] 2733  quiescing stats garbage collector
I241031 08:50:16.590722 9637 server/start_listen.go:111  [T1,Vsystem,n1] 2734  server shutting down: instructing cmux to stop accepting
W241031 08:50:16.601519 45354 jobs/adopt.go:490  [T1,Vsystem,n1] 2735  could not clear job claim: clear-job-claim: node unavailable; try another peer
W241031 08:50:16.602583 45348 jobs/adopt.go:490  [T1,Vsystem,n1] 2736  could not clear job claim: clear-job-claim: node unavailable; try another peer
W241031 08:50:16.602800 10061 sql/sqlliveness/slinstance/slinstance.go:336  [T1,Vsystem,n1] 2737  exiting heartbeat loop
W241031 08:50:16.602862 10061 sql/sqlliveness/slinstance/slinstance.go:323  [T1,Vsystem,n1] 2738  exiting heartbeat loop with error: node unavailable; try another peer
E241031 08:50:16.602901 10061 server/server_sql.go:515  [T1,Vsystem,n1] 2739  failed to run update of instance with new session ID: node unavailable; try another peer
I241031 08:50:16.603098 10481 jobs/registry.go:1598  [T1,Vsystem,n1] 2740  AUTO SPAN CONFIG RECONCILIATION job 1016771929412599809: stepping through state succeeded
E241031 08:50:16.603301 10481 jobs/registry.go:897  [T1,Vsystem,n1] 2741  error getting live session: node unavailable; try another peer
I241031 08:50:16.603681 12487 4@util/log/event_log.go:39  [T10,Vsrc,n1,client=127.0.0.1:35510,hostssl,user=root] 2742 ={"Timestamp":1730364616603677120,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:35510","SessionID":"18037d6868d6df8b0000000000000001","Duration":47374289959}
W241031 08:50:16.604122 12013 server/server_sql.go:1780  [T11,Vdest,n1] 2743  server shutdown without a prior graceful drain
I241031 08:50:16.604183 12013 1@util/log/event_log.go:39  [n1,tenant-orchestration,tenant=dest] 2744 ={"Timestamp":1730364616604181790,"EventType":"tenant_shared_service_stop","NodeID":1,"TenantID":11,"InstanceID":1,"TenantName":"dest"}
W241031 08:50:16.604241 12013 sql/event_log.go:658  [n1,tenant-orchestration,tenant=dest] 2745  failed to start task to save 1 events in eventlog: node unavailable; try another peer
I241031 08:50:16.604605 12013 server/server_controller_channel_orchestrator.go:486  [n1,tenant-orchestration,tenant=dest] 2746  propagate-ungraceful-stop task terminating
I241031 08:50:16.604676 12011 server/server_controller_channel_orchestrator.go:286  [T1,Vsystem,n1] 2747  propagate-close task terminating
I241031 08:50:16.604861 15109 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:35522,hostssl,user=root] 2748 ={"Timestamp":1730364616604859710,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:35522","SessionID":"18037d698f1212e00000000000000001","Duration":42439091380}
W241031 08:50:16.620659 11384 server/server_sql.go:1780  [T10,Vsrc,n1] 2749  server shutdown without a prior graceful drain
I241031 08:50:16.620797 11384 1@util/log/event_log.go:39  [n1,tenant-orchestration,tenant=src] 2750 ={"Timestamp":1730364616620794109,"EventType":"tenant_shared_service_stop","NodeID":1,"TenantID":10,"InstanceID":1,"TenantName":"src"}
W241031 08:50:16.620903 11384 sql/event_log.go:658  [n1,tenant-orchestration,tenant=src] 2751  failed to start task to save 1 events in eventlog: node unavailable; try another peer
I241031 08:50:16.621321 11384 server/server_controller_channel_orchestrator.go:486  [n1,tenant-orchestration,tenant=src] 2752  propagate-ungraceful-stop task terminating
I241031 08:50:16.637317 11382 server/server_controller_channel_orchestrator.go:286  [T1,Vsystem,n1] 2753  propagate-close task terminating
I241031 08:50:16.638081 9560 server/server_controller.go:409  [T1,Vsystem,n1] 2754  server controller shutting down
I241031 08:50:16.638144 9560 server/server_controller.go:418  [T1,Vsystem,n1] 2755  waiting for tenant servers to report stopped
W241031 08:50:16.638250 9560 server/server_sql.go:1780  [T1,Vsystem,n1] 2756  server shutdown without a prior graceful drain
--- FAIL: TestReaderCatalogTSAdvance (48.94s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #133902 pkg/sql/catalog/replication/replication_test: TestReaderCatalogTSAdvance failed [C-test-failure O-robot T-sql-foundations branch-release-24.3 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://tanzanite.cluster.engflow.com/invocations/default/425a7a01-acd0-420b-be1c-42c11e80b412?testReportRun=23&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvcmVwbGljYXRpb246cmVwbGljYXRpb25fdGVzdA==) on master @ [482ba56b86710a85c3d5e069d6849edce2472b84](https://github.com/cockroachdb/cockroach/commits/482ba56b86710a85c3d5e069d6849edce2472b84):


```
I241101 05:17:31.426294 11587 server/server_controller_channel_orchestrator.go:442  [n1,tenant-orchestration,tenant=dest] 2734  tenant "dest" finishing their own control loop
I241101 05:17:31.426525 11628 server/start_listen.go:111  [T11,Vdest,n1] 2735  server shutting down: instructing cmux to stop accepting
I241101 05:17:31.427003 11803 jobs/registry.go:1598  [T11,Vdest,n1] 2736  AUTO SPAN CONFIG RECONCILIATION job 1017013221230247937: stepping through state succeeded
W241101 05:17:31.427388 11803 jobs/adopt.go:490  [T11,Vdest,n1] 2737  could not clear job claim: clear-job-claim: node unavailable; try another peer
W241101 05:17:31.427899 38839 jobs/adopt.go:490  [T11,Vdest,n1] 2738  could not clear job claim: clear-job-claim: node unavailable; try another peer
W241101 05:17:31.428242 11631 kv/kvclient/kvtenant/connector.go:424  [T11,Vdest,n1,tenant-connector] 2739  error consuming GossipSubscription RPC: recv msg error: grpc: context canceled [code 1/Canceled]
W241101 05:17:31.428397 11632 kv/kvclient/kvtenant/setting_overrides.go:65  [T11,Vdest,n1,tenant-connector] 2740  error consuming TenantSettings RPC: recv msg error: grpc: context canceled [code 1/Canceled]
W241101 05:17:31.428702 11674 sql/sqlliveness/slinstance/slinstance.go:336  [T11,Vdest,n1] 2741  exiting heartbeat loop
W241101 05:17:31.428753 11674 sql/sqlliveness/slinstance/slinstance.go:323  [T11,Vdest,n1] 2742  exiting heartbeat loop with error: node unavailable; try another peer
E241101 05:17:31.428790 11674 server/server_sql.go:515  [T11,Vdest,n1] 2743  failed to run update of instance with new session ID: node unavailable; try another peer
E241101 05:17:31.428921 38834 jobs/registry.go:897  [T11,Vdest,n1] 2744  error getting live session: node unavailable; try another peer
I241101 05:17:31.429134 47157 jobs/registry.go:1598  [T10,Vsrc,n1] 2745  AUTO UPDATE SQL ACTIVITY job 103: stepping through state succeeded
W241101 05:17:31.429492 47157 jobs/adopt.go:490  [T10,Vsrc,n1] 2746  could not clear job claim: clear-job-claim: node unavailable; try another peer
I241101 05:17:31.429587 11394 server/server_controller_channel_orchestrator.go:502  [n1,tenant-orchestration,tenant=src] 2747  propagate-graceful-drain task terminating
I241101 05:17:31.429663 11387 sql/stats/automatic_stats.go:712  [T10,Vsrc,n1] 2748  quiescing stats garbage collector
I241101 05:17:31.429711 10959 server/server_controller_channel_orchestrator.go:442  [n1,tenant-orchestration,tenant=src] 2749  tenant "src" finishing their own control loop
I241101 05:17:31.429878 11000 server/start_listen.go:111  [T10,Vsrc,n1] 2750  server shutting down: instructing cmux to stop accepting
W241101 05:17:31.430656 47165 jobs/adopt.go:490  [T10,Vsrc,n1] 2751  could not clear job claim: clear-job-claim: node unavailable; try another peer
I241101 05:17:31.430964 11171 jobs/registry.go:1598  [T10,Vsrc,n1] 2752  AUTO SPAN CONFIG RECONCILIATION job 1017013219652698113: stepping through state succeeded
W241101 05:17:31.431324 11171 jobs/adopt.go:490  [T10,Vsrc,n1] 2753  could not clear job claim: clear-job-claim: node unavailable; try another peer
W241101 05:17:31.431450 11004 kv/kvclient/kvtenant/setting_overrides.go:65  [T10,Vsrc,n1,tenant-connector] 2754  error consuming TenantSettings RPC: recv msg error: grpc: node unavailable; try another peer [code 2/Unknown]
W241101 05:17:31.431697 11061 sql/sqlliveness/slinstance/slinstance.go:336  [T10,Vsrc,n1] 2755  exiting heartbeat loop
W241101 05:17:31.431761 11061 sql/sqlliveness/slinstance/slinstance.go:323  [T10,Vsrc,n1] 2756  exiting heartbeat loop with error: node unavailable; try another peer
E241101 05:17:31.431804 11061 server/server_sql.go:515  [T10,Vsrc,n1] 2757  failed to run update of instance with new session ID: node unavailable; try another peer
W241101 05:17:31.431870 11003 kv/kvclient/kvtenant/connector.go:424  [T10,Vsrc,n1,tenant-connector] 2758  error consuming GossipSubscription RPC: recv msg error: grpc: node unavailable; try another peer [code 2/Unknown]
E241101 05:17:31.432189 47172 jobs/registry.go:897  [T10,Vsrc,n1] 2759  error getting live session: node unavailable; try another peer
I241101 05:17:31.432262 10958 server/server_controller_channel_orchestrator.go:261  [T1,Vsystem,n1] 2760  server terminating; telling tenant "src" to terminate
I241101 05:17:31.432459 10539 sql/stats/automatic_stats.go:712  [T1,Vsystem,n1] 2761  quiescing stats garbage collector
I241101 05:17:31.432736 9212 server/start_listen.go:111  [T1,Vsystem,n1] 2762  server shutting down: instructing cmux to stop accepting
W241101 05:17:31.433123 9638 sql/sqlliveness/slinstance/slinstance.go:336  [T1,Vsystem,n1] 2763  exiting heartbeat loop
W241101 05:17:31.433176 9638 sql/sqlliveness/slinstance/slinstance.go:323  [T1,Vsystem,n1] 2764  exiting heartbeat loop with error: node unavailable; try another peer
E241101 05:17:31.433223 9638 server/server_sql.go:515  [T1,Vsystem,n1] 2765  failed to run update of instance with new session ID: node unavailable; try another peer
E241101 05:17:31.433345 41760 jobs/registry.go:897  [T1,Vsystem,n1] 2766  error getting live session: node unavailable; try another peer
I241101 05:17:31.434017 10101 jobs/registry.go:1598  [T1,Vsystem,n1] 2767  AUTO SPAN CONFIG RECONCILIATION job 1017013217583235073: stepping through state succeeded
I241101 05:17:31.434701 12055 4@util/log/event_log.go:39  [T10,Vsrc,n1,client=127.0.0.1:50108,hostssl,user=root] 2768 ={"Timestamp":1730438251434698264,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:50108","SessionID":"1803c060f607a4570000000000000001","Duration":46917204885}
W241101 05:17:31.435010 11588 server/server_sql.go:1780  [T11,Vdest,n1] 2769  server shutdown without a prior graceful drain
I241101 05:17:31.435075 11588 1@util/log/event_log.go:39  [n1,tenant-orchestration,tenant=dest] 2770 ={"Timestamp":1730438251435073825,"EventType":"tenant_shared_service_stop","NodeID":1,"TenantID":11,"InstanceID":1,"TenantName":"dest"}
W241101 05:17:31.435136 11588 sql/event_log.go:658  [n1,tenant-orchestration,tenant=dest] 2771  failed to start task to save 1 events in eventlog: node unavailable; try another peer
I241101 05:17:31.435492 11588 server/server_controller_channel_orchestrator.go:486  [n1,tenant-orchestration,tenant=dest] 2772  propagate-ungraceful-stop task terminating
I241101 05:17:31.435732 14622 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:50118,hostssl,user=root] 2773 ={"Timestamp":1730438251435729949,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:50118","SessionID":"1803c0621b862b910000000000000001","Duration":41994220358}
I241101 05:17:31.435884 11586 server/server_controller_channel_orchestrator.go:286  [T1,Vsystem,n1] 2774  propagate-close task terminating
W241101 05:17:31.435987 10960 server/server_sql.go:1780  [T10,Vsrc,n1] 2775  server shutdown without a prior graceful drain
I241101 05:17:31.436044 10960 1@util/log/event_log.go:39  [n1,tenant-orchestration,tenant=src] 2776 ={"Timestamp":1730438251436043521,"EventType":"tenant_shared_service_stop","NodeID":1,"TenantID":10,"InstanceID":1,"TenantName":"src"}
W241101 05:17:31.436104 10960 sql/event_log.go:658  [n1,tenant-orchestration,tenant=src] 2777  failed to start task to save 1 events in eventlog: node unavailable; try another peer
I241101 05:17:31.436413 10960 server/server_controller_channel_orchestrator.go:486  [n1,tenant-orchestration,tenant=src] 2778  propagate-ungraceful-stop task terminating
I241101 05:17:31.448511 10958 server/server_controller_channel_orchestrator.go:286  [T1,Vsystem,n1] 2779  propagate-close task terminating
I241101 05:17:31.464292 9135 server/server_controller.go:409  [T1,Vsystem,n1] 2780  server controller shutting down
I241101 05:17:31.464362 9135 server/server_controller.go:418  [T1,Vsystem,n1] 2781  waiting for tenant servers to report stopped
W241101 05:17:31.464457 9135 server/server_sql.go:1780  [T1,Vsystem,n1] 2782  server shutdown without a prior graceful drain
--- FAIL: TestReaderCatalogTSAdvance (48.40s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=23</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #133902 pkg/sql/catalog/replication/replication_test: TestReaderCatalogTSAdvance failed [C-test-failure O-robot T-sql-foundations branch-release-24.3]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17546743?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17546743?buildTab=artifacts#/) on master @ [e7588167d28a5c6e99861df117caa6a83d88f208](https://github.com/cockroachdb/cockroach/commits/e7588167d28a5c6e99861df117caa6a83d88f208):

Fatal error:

```
panic: test timed out after 5m0s
running tests:
	TestReaderCatalogTSAdvance (4m35s)

```
Stack: 

```
goroutine 158378 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
I241101 13:07:00.842292 95658 sql/catalog/lease/lease.go:273  [-] 3944  waiting for 1 leases to expire: desc=[{users 4 121}]
I241101 13:07:00.848747 153318 sql/catalog/lease/descriptor_state.go:142  [n1] 3945  new lease: tabledesc.immutable: {ID: 107, Version: 121, ModificationTime: "1730466420.690668995,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I241101 13:07:00.864222 153323 sql/catalog/lease/descriptor_state.go:142  [n1] 3946  new lease: tabledesc.immutable: {ID: 109, Version: 121, ModificationTime: "1730466420.690668995,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I241101 13:07:02.259201 95658 sql/catalog/lease/lease.go:273  [-] 3947  waiting for 1 leases to expire: desc=[{users 4 122}]
I241101 13:07:02.268901 154167 sql/catalog/lease/descriptor_state.go:142  [n1] 3948  new lease: tabledesc.immutable: {ID: 4, Version: 123, ModificationTime: "1730466422.196120233,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I241101 13:07:02.270741 154138 sql/catalog/lease/descriptor_state.go:142  [n1] 3949  new lease: tabledesc.immutable: {ID: 20, Version: 123, ModificationTime: "1730466422.196120233,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I241101 13:07:02.283680 153960 sql/catalog/lease/descriptor_state.go:142  [n1] 3950  new lease: tabledesc.immutable: {ID: 23, Version: 123, ModificationTime: "1730466422.196120233,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I241101 13:07:02.306946 95658 sql/catalog/lease/lease.go:273  [-] 3951  waiting for 1 leases to expire: desc=[{table_statistics 20 122}]
I241101 13:07:02.312099 154149 sql/catalog/lease/descriptor_state.go:142  [n1] 3952  new lease: tabledesc.immutable: {ID: 33, Version: 123, ModificationTime: "1730466422.196120233,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I241101 13:07:02.328084 154145 sql/catalog/lease/descriptor_state.go:142  [n1] 3953  new lease: tabledesc.immutable: {ID: 44, Version: 123, ModificationTime: "1730466422.196120233,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I241101 13:07:02.331347 154213 sql/catalog/lease/descriptor_state.go:142  [n1] 3954  new lease: tabledesc.immutable: {ID: 107, Version: 122, ModificationTime: "1730466422.196120233,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I241101 13:07:02.332780 95658 sql/catalog/lease/lease.go:273  [-] 3955  waiting for 1 leases to expire: desc=[{database_role_settings 44 122}]
I241101 13:07:02.333503 154188 sql/catalog/lease/descriptor_state.go:142  [n1] 3956  new lease: tabledesc.immutable: {ID: 109, Version: 122, ModificationTime: "1730466422.196120233,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I241101 13:07:02.556307 11061 kv/kvserver/queue.go:616  [T1,Vsystem,n1,s1,r57/1:/Table/5{5-6},raft] 3957  rate limited in MaybeAdd (merge): throttled on async limiting semaphore
I241101 13:07:03.810281 95658 sql/catalog/lease/lease.go:273  [-] 3958  waiting for 1 leases to expire: desc=[{users 4 123}]
I241101 13:07:03.856161 155252 sql/catalog/lease/descriptor_state.go:142  [n1] 3959  new lease: tabledesc.immutable: {ID: 4, Version: 124, ModificationTime: "1730466423.733092456,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I241101 13:07:03.857969 155257 sql/catalog/lease/descriptor_state.go:142  [n1] 3960  new lease: tabledesc.immutable: {ID: 20, Version: 124, ModificationTime: "1730466423.733092456,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I241101 13:07:03.860412 155229 sql/catalog/lease/descriptor_state.go:142  [n1] 3961  new lease: tabledesc.immutable: {ID: 23, Version: 124, ModificationTime: "1730466423.733092456,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I241101 13:07:03.863115 155267 sql/catalog/lease/descriptor_state.go:142  [n1] 3962  new lease: tabledesc.immutable: {ID: 33, Version: 124, ModificationTime: "1730466423.733092456,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I241101 13:07:03.866711 155272 sql/catalog/lease/descriptor_state.go:142  [n1] 3963  new lease: tabledesc.immutable: {ID: 44, Version: 124, ModificationTime: "1730466423.733092456,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I241101 13:07:03.868737 155262 sql/catalog/lease/descriptor_state.go:142  [n1] 3964  new lease: tabledesc.immutable: {ID: 107, Version: 123, ModificationTime: "1730466423.733092456,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I241101 13:07:03.884281 155283 sql/catalog/lease/descriptor_state.go:142  [n1] 3965  new lease: tabledesc.immutable: {ID: 109, Version: 123, ModificationTime: "1730466423.733092456,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I241101 13:07:04.661384 11189 2@util/log/event_log.go:39  [T1,Vsystem,n1] 3966 ={"Timestamp":1730466424661379875,"EventType":"runtime_stats","MemRSSBytes":699895808,"GoroutineCount":1279,"MemStackSysBytes":11042816,"GoAllocBytes":285291008,"GoTotalBytes":504472968,"HeapFragmentBytes":107949568,"HeapReservedBytes":78290944,"HeapReleasedBytes":83566592,"CGoAllocBytes":146305840,"CGoTotalBytes":202375168,"CGoCallRate":23.847822,"CPUUserPercent":93.295876,"CPUSysPercent":6.5855913,"GCPausePercent":0.8060904,"GCRunCount":172,"NetHostRecvBytes":122734527,"NetHostSendBytes":122728652}
I241101 13:07:04.661503 11189 2@server/status/runtime_log.go:43  [T1,Vsystem,n1] 3967  runtime stats: 668 MiB RSS, 1279 goroutines (stacks: 10 MiB), 272 MiB/481 MiB Go alloc/total (heap fragmentation: 103 MiB, heap reserved: 75 MiB, heap released: 80 MiB), 140 MiB/193 MiB CGO alloc/total (23.8 CGO/sec), 93.3/6.6 %(u/s)time, 0.8 %gc (172x), 117 MiB/117 MiB (r/w)net
I241101 13:07:05.960528 155967 sql/catalog/lease/descriptor_state.go:142  [n1] 3968  new lease: tabledesc.immutable: {ID: 4, Version: 125, ModificationTime: "1730466425.899301042,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I241101 13:07:05.963614 156197 sql/catalog/lease/descriptor_state.go:142  [n1] 3969  new lease: tabledesc.immutable: {ID: 20, Version: 125, ModificationTime: "1730466425.899301042,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I241101 13:07:05.966036 156202 sql/catalog/lease/descriptor_state.go:142  [n1] 3970  new lease: tabledesc.immutable: {ID: 23, Version: 125, ModificationTime: "1730466425.899301042,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I241101 13:07:05.976641 155311 sql/catalog/lease/descriptor_state.go:142  [n1] 3971  new lease: tabledesc.immutable: {ID: 33, Version: 125, ModificationTime: "1730466425.899301042,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I241101 13:07:05.977332 95658 sql/catalog/lease/lease.go:273  [-] 3972  waiting for 1 leases to expire: desc=[{users 4 124}]
I241101 13:07:05.992286 156211 sql/catalog/lease/descriptor_state.go:142  [n1] 3973  new lease: tabledesc.immutable: {ID: 44, Version: 125, ModificationTime: "1730466425.899301042,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I241101 13:07:05.994938 95658 sql/catalog/lease/lease.go:273  [-] 3974  waiting for 1 leases to expire: desc=[{database_role_settings 44 124}]
I241101 13:07:05.995450 156216 sql/catalog/lease/descriptor_state.go:142  [n1] 3975  new lease: tabledesc.immutable: {ID: 107, Version: 124, ModificationTime: "1730466425.899301042,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I241101 13:07:05.997810 156221 sql/catalog/lease/descriptor_state.go:142  [n1] 3976  new lease: tabledesc.immutable: {ID: 109, Version: 124, ModificationTime: "1730466425.899301042,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I241101 13:07:07.676069 157063 sql/catalog/lease/descriptor_state.go:142  [n1] 3977  new lease: tabledesc.immutable: {ID: 4, Version: 126, ModificationTime: "1730466427.583731683,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I241101 13:07:07.679014 157208 sql/catalog/lease/descriptor_state.go:142  [n1] 3978  new lease: tabledesc.immutable: {ID: 20, Version: 126, ModificationTime: "1730466427.583731683,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I241101 13:07:07.685193 157166 sql/catalog/lease/descriptor_state.go:142  [n1] 3979  new lease: tabledesc.immutable: {ID: 23, Version: 126, ModificationTime: "1730466427.583731683,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I241101 13:07:07.687535 156675 sql/catalog/lease/descriptor_state.go:142  [n1] 3980  new lease: tabledesc.immutable: {ID: 33, Version: 126, ModificationTime: "1730466427.583731683,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I241101 13:07:07.689253 95658 sql/catalog/lease/lease.go:273  [-] 3981  waiting for 1 leases to expire: desc=[{users 4 125}]
I241101 13:07:07.689710 156680 sql/catalog/lease/descriptor_state.go:142  [n1] 3982  new lease: tabledesc.immutable: {ID: 44, Version: 126, ModificationTime: "1730466427.583731683,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I241101 13:07:07.702386 157251 sql/catalog/lease/descriptor_state.go:142  [n1] 3983  new lease: tabledesc.immutable: {ID: 107, Version: 125, ModificationTime: "1730466427.583731683,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I241101 13:07:07.704591 157256 sql/catalog/lease/descriptor_state.go:142  [n1] 3984  new lease: tabledesc.immutable: {ID: 109, Version: 125, ModificationTime: "1730466427.583731683,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I241101 13:07:09.051443 11098 kv/kvserver/queue.go:616  [T1,Vsystem,n1,s1,r91/1:/Tenant/10/Table/4{7-8},raft] 3985  rate limited in MaybeAdd (merge): throttled on async limiting semaphore
I241101 13:07:09.167256 158155 sql/catalog/lease/descriptor_state.go:142  [n1] 3986  new lease: tabledesc.immutable: {ID: 4, Version: 127, ModificationTime: "1730466429.110141722,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I241101 13:07:09.169674 158160 sql/catalog/lease/descriptor_state.go:142  [n1] 3987  new lease: tabledesc.immutable: {ID: 20, Version: 127, ModificationTime: "1730466429.110141722,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I241101 13:07:09.172522 158261 sql/catalog/lease/descriptor_state.go:142  [n1] 3988  new lease: tabledesc.immutable: {ID: 23, Version: 127, ModificationTime: "1730466429.110141722,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I241101 13:07:09.175205 158267 sql/catalog/lease/descriptor_state.go:142  [n1] 3989  new lease: tabledesc.immutable: {ID: 33, Version: 127, ModificationTime: "1730466429.110141722,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I241101 13:07:09.177070 158272 sql/catalog/lease/descriptor_state.go:142  [n1] 3990  new lease: tabledesc.immutable: {ID: 44, Version: 127, ModificationTime: "1730466429.110141722,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I241101 13:07:09.179044 158277 sql/catalog/lease/descriptor_state.go:142  [n1] 3991  new lease: tabledesc.immutable: {ID: 107, Version: 126, ModificationTime: "1730466429.110141722,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I241101 13:07:09.180854 158200 sql/catalog/lease/descriptor_state.go:142  [n1] 3992  new lease: tabledesc.immutable: {ID: 109, Version: 126, ModificationTime: "1730466429.110141722,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I241101 13:07:09.188452 95658 sql/catalog/lease/lease.go:273  [-] 3993  waiting for 1 leases to expire: desc=[{users 4 126}]
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #133902 pkg/sql/catalog/replication/replication_test: TestReaderCatalogTSAdvance failed [C-test-failure O-robot T-sql-foundations branch-release-24.3]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/catalog/replication/BUILD.bazel
+++ b/pkg/sql/catalog/replication/BUILD.bazel
@@ -35,6 +35,7 @@ go_test(
         "//pkg/security/securitytest",
         "//pkg/server",
         "//pkg/sql",
+        "//pkg/sql/catalog/descpb",
         "//pkg/sql/catalog/lease",
         "//pkg/testutils",
         "//pkg/testutils/serverutils",

--- a/pkg/sql/catalog/replication/reader_catalog_test.go
+++ b/pkg/sql/catalog/replication/reader_catalog_test.go
@@ -18,6 +18,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/security/securitytest"
 	"github.com/cockroachdb/cockroach/pkg/server"
 	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/lease"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/replication"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
@@ -236,9 +237,30 @@ func TestReaderCatalogTSAdvance(t *testing.T) {
 		TenantName: "src",
 	})
 	require.NoError(t, err)
+
+	waitForRefresh := make(chan struct{})
+	descriptorRefreshHookEnabled := atomic.Bool{}
+	closeWaitForRefresh := func() {
+		if descriptorRefreshHookEnabled.Load() {
+			descriptorRefreshHookEnabled.Store(false)
+			close(waitForRefresh)
+		}
+	}
+	defer closeWaitForRefresh()
+	destTestingKnobs := base.TestingKnobs{
+		SQLLeaseManager: &lease.ManagerTestingKnobs{
+			TestingDescriptorRefreshedEvent: func(descriptor *descpb.Descriptor) {
+				if !descriptorRefreshHookEnabled.Load() {
+					return
+				}
+				<-waitForRefresh
+			},
+		},
+	}
 	destTenant, _, err := ts.StartSharedProcessTenant(ctx, base.TestSharedProcessTenantArgs{
 		TenantID:   serverutils.TestTenantID2(),
 		TenantName: "dest",
+		Knobs:      destTestingKnobs,
 	})
 	require.NoError(t, err)
 	srcConn := srcTenant.SQLConn(t)
@@ -307,8 +329,14 @@ func TestReaderCatalogTSAdvance(t *testing.T) {
 	compareEqual("SELECT * FROM t1")
 
 	var newTS hlc.Timestamp
-
+	descriptorRefreshHookEnabled.Store(true)
 	for _, useAOST := range []bool{false, true} {
+		if useAOST {
+			closeWaitForRefresh()
+		}
+		// When AOST is enabled then a fixed number of iterations sufficient. When
+		// the AOST is disabled, then we will iterate until a reader timestamp error
+		// is generated.
 		errorDetected := false
 		checkAOSTError := func(err error) {
 			if useAOST || err == nil {
@@ -323,12 +351,14 @@ func TestReaderCatalogTSAdvance(t *testing.T) {
 		// are not line up on the reader catalog.
 		grp := ctxgroup.WithContext(ctx)
 		require.NoError(t, err)
-		var iterationsDone atomic.Bool
+		iterationsDoneCh := make(chan struct{})
 		grp.GoCtx(func(ctx context.Context) error {
 			defer func() {
-				iterationsDone.Swap(true)
+				close(iterationsDoneCh)
 			}()
-			const NumIterations = 64
+			const NumIterations = 16
+			// Ensure the minimum iterations are met, and any expected errors
+			// are observed before stopping TS advances.
 			for iter := 0; iter < NumIterations; iter++ {
 				if _, err := srcRunner.DB.ExecContext(ctx,
 					"INSERT INTO t1(val, j) VALUES('open', $1);",
@@ -351,22 +381,35 @@ func TestReaderCatalogTSAdvance(t *testing.T) {
 		} else {
 			destRunner.Exec(t, "SET bypass_pcr_reader_catalog_aost='on'")
 		}
-		for !iterationsDone.Load() {
-			tx := destRunner.Begin(t)
-			_, err := tx.Exec("SELECT * FROM t1")
-			checkAOSTError(err)
-			_, err = tx.Exec("SELECT * FROM v1")
-			checkAOSTError(err)
-			_, err = tx.Exec("SELECT * FROM t2")
-			checkAOSTError(err)
-			checkAOSTError(tx.Commit())
+		iterationsDone := false
+		for !iterationsDone {
+			if !useAOST {
+				select {
+				case waitForRefresh <- struct{}{}:
+				case <-iterationsDoneCh:
+					iterationsDone = true
+				}
+			}
+			select {
+			case <-iterationsDoneCh:
+				iterationsDone = true
+			default:
+				tx := destRunner.Begin(t)
+				_, err := tx.Exec("SELECT * FROM t1")
+				checkAOSTError(err)
+				_, err = tx.Exec("SELECT * FROM v1")
+				checkAOSTError(err)
+				_, err = tx.Exec("SELECT * FROM t2")
+				checkAOSTError(err)
+				checkAOSTError(tx.Commit())
 
-			_, err = destRunner.DB.ExecContext(ctx, "SELECT * FROM t1,v1, t2")
-			checkAOSTError(err)
-			_, err = destRunner.DB.ExecContext(ctx, "SELECT * FROM v1 ORDER BY 1")
-			checkAOSTError(err)
-			_, err = destRunner.DB.ExecContext(ctx, "SELECT * FROM t2 ORDER BY 1")
-			checkAOSTError(err)
+				_, err = destRunner.DB.ExecContext(ctx, "SELECT * FROM t1,v1, t2")
+				checkAOSTError(err)
+				_, err = destRunner.DB.ExecContext(ctx, "SELECT * FROM v1 ORDER BY 1")
+				checkAOSTError(err)
+				_, err = destRunner.DB.ExecContext(ctx, "SELECT * FROM t2 ORDER BY 1")
+				checkAOSTError(err)
+			}
 		}
 
 		// Finally ensure the queries actually match.

